synapse/handlers/sync.py

```diff
@@ -574,8 +574,11 @@ async def _load_filtered_recents(
                 # ensure that we always include current state in the timeline
                 current_state_ids = frozenset()
                 if any(e.is_state() for e in loaded_recents):
-                    current_state_ids_map = await self.store.get_current_state_ids(
-                        room_id
+                    # FIXME(faster room joins): We use the partial state here as
+                    # we don't want to block `/sync` on finishing a lazy join.
+                    # Is this the correct way of doing it?
+                    current_state_ids_map = (
+                        await self.store.get_partial_current_state_ids(room_id)
```

Comment:
This is a check to see if the state event we're about to send down to clients is in the current state, and if so to always send it even if `filter_events_for_client` would filter it out. Not sure if using `get_partial_current_state_ids` is the right approach here?